comp/dogstatsd/server/server.go

```diff
@@ -497,16 +501,31 @@ func (s *server) SetExtraTags(tags []string) {
 	s.extraTags = tags
 }
 
+// SetBlocklist updates the metric names blocklist on all running worker.
+func (s *server) SetBlocklist(metricsName []string, matchPrefix bool) {
+	// each worker receives its own copy
+	for _, worker := range s.workers {
+		blocklist := newBlocklist(metricsName, matchPrefix)
```

Comment:
blocklist can be shared between workers. If `strings.Sort` inside `newBlocklist` temporarily modifies the slice even if it is sorted, it may cause data race or briefly break previous instances of blocklist using the same slice.

Reply:
That's why I'm creating a copy of metricNames in `newBlocklist`, am I missing something?